.github/workflows/reusable_testing.yml

```diff
@@ -0,0 +1,48 @@
+name: reusable_testing
+
+on:
+  workflow_call:
+    inputs:
+      artifact_name:
+        description: 'The name of the tarantool build artifact'
+        default: ubuntu-focal
+        required: false
+        type: string
+
+jobs:
+  run_tests:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: 'Clone the smtp module'
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ github.repository_owner }}/smtp
+
+      - name: 'Download the tarantool build artifact'
+        uses: actions/download-artifact@v2
+        with:
+          name: ${{ inputs.artifact_name }}
+      # TODO(ylobankov): Just in case remove tarantool that could be previously
+      # installed on the runner. Relevant to self-hosted runners. Remove this
+      # step when only GitHub-hosted runners are used for integration testing.
```

Comment:
It'll spread across ~20 repositories soon? And, if we'll need to adjust those workarounds, you'll open ~20 PRs? And wait for maintainers to look at them and push?

Can we spend a day of work and pin those runs to GitHub hosted runners? It'll be around 20 seconds or even less: almost nothing comparing to heavy tarantool's jobs.

Reply:
Have we decided that this is the best solution now? Are we definitely not changing it after a near period?
I'd like to understand this point and fix it for the future.

Reply:
It looks as the best now, because reduces manual clean up activities as well as effort to automate them. What about a future: I hope, if we'll pay more effort to our own infra, we'll move toward autocreation of clean virtual machines. Otherwise, sooner or later, we'll completely miss understanding of the environment, where our workflows run.

Even if we'll return back here in a future and will want to run this workflow on self-hosted machines, I hope, they will not require such clean up.

Reply:
Opened related PR https://github.com/tarantool/tarantool/pull/6537.